### PR TITLE
fixes the cli preset flag to work properly

### PR DESCRIPTION
**Pull request by FredKSchott**
- [X] Changelog updated
- Test added
- Did some small refactoring to go from 1 build integration test to N integration tests

### CODE REVIEW

File: src/commands/build.ts [hunk 1]
```diff
@@ -135,6 +105,22 @@ export class BuildCommand implements Command {
...
   ];
 
+  /**
+   * Converts command-line build arguments to the `ProjectBuildOptions` format
+   * that our build understands, applying the preset if one was given.
+   */
+  private commandOptionsToBuildOptions(options: CommandOptions):
+      ProjectBuildOptions {
+    const buildOptions: ProjectBuildOptions = {};
+    const validBuildOptions = this.args.map(({name}) => name);
```
Comment (justinfagnani) [inline src/commands/build.ts]:
```ts
const validBuildOptions = new Set(this.args.map(({name}) => name));
```

File: src/commands/build.ts [hunk 2]
```diff
@@ -135,6 +105,22 @@ export class BuildCommand implements Command {
...
+    for (const buildOption in options) {
```
Comment (justinfagnani) [inline src/commands/build.ts]:
```ts
for (const buildOption in Object.keys(options)) {
```

and remove the hasOwnProperty check below

File: src/commands/build.ts [hunk 3]
```diff
@@ -24,40 +24,10 @@ import {PolymerProject} from 'polymer-build';
...
-
 const logger = logging.getLogger('cli.command.build');
 
-/**
- * Converts command-line build arguments to the `ProjectBuildOptions` format
- * that our build understands, applying the preset if one was given.
- */
-function commandOptionsToBuildOptions(options: CommandOptions):
-    ProjectBuildOptions {
-  return applyBuildPreset({
```
Comment (justinfagnani) [inline src/commands/build.ts]: where was the actual bug here?

Comment (FredKSchott) [inline src/commands/build.ts]: By defining all options in the argument, `applyBuildPreset()` interpreted them all as overrides on the preset.

**Pull request merged**